File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    
    const obj = {};
    
    for (const item of clothes) {
        if (obj[item[1]]) obj[item[1]].push(item[0]);
        else obj[item[1]] = [item[0]];
    }
    
    Object.values(obj).forEach((item) => {
        answer *= (item.length + 1)
    })
    
    return answer - 1;
}

File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    obj = {}
    
    for cloth in clothes:
        [item, item_type] = cloth
        if item_type in obj:
            obj[item_type].append(item)
        else:
            obj[item_type] = [item]
    
    for item in obj:
        answer *= len(obj[item]) + 1
        
    return answer - 1